modules/Emails/EmailsController.php

```diff
@@ -627,4 +651,49 @@ protected function setAfterImport($importedEmailId, $request)
 
         return $emails;
     }
+
+    /**
+     * @param User $requestedUser
+     * @param InboundEmail $requestedInboundEmail
+     * @param Email $requestedEmail
+     * @return bool false if user doesn't have access
+     */
+    protected function userIsAllowedToSendEmail($requestedUser,  $requestedInboundEmail, $requestedEmail)
+    {
+        $hasAccess = false;
+        $hasAccessToInboundEmailAccount = false;
+
+        // Check that user is allowed to use inbound email account
+        $usersInboundEmailAccounts = $requestedInboundEmail->retrieveByGroupId($requestedUser->id);
+        foreach ($usersInboundEmailAccounts as $inboundEmailId => $userInboundEmail) {
+            if($userInboundEmail->id === $requestedInboundEmail->id) {
+                $hasAccessToInboundEmailAccount = true;
+                break;
+            }
+        }
+
```

Comment:
You can return early if `$hasAccessToInboundEmailAccount` is false, to avoid extra work, since this code forces the existence of an inbound account (ref #3741)